src/Stratis.Bitcoin.Features.SignalR/Broadcasters/BroadcasterBase.cs

```diff
@@ -0,0 +1,57 @@
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Stratis.Bitcoin.AsyncWork;
+using Stratis.Bitcoin.Signals;
+using Stratis.Bitcoin.Utilities;
+
+namespace Stratis.Bitcoin.Features.SignalR.Broadcasters
+{
+    public abstract class ClientBroadcasterBase : IClientEventBroadcaster
+    {
+        private readonly EventsHub eventsHub;
+        private readonly INodeLifetime nodeLifetime;
+        private readonly ILoggerFactory loggerFactory;
+        private readonly AsyncProvider asyncProvider;
+        private readonly int broadcastFrequencySeconds;
+        protected readonly ILogger logger;
+        private IAsyncLoop asyncLoop;
+
+
+        protected ClientBroadcasterBase(
+            EventsHub eventsHub,
+            ISignals signals,
+            INodeLifetime nodeLifetime,
+            ILoggerFactory loggerFactory,
+            int broadcastFrequencySeconds = 5)
+        {
+            this.eventsHub = eventsHub;
+            this.nodeLifetime = nodeLifetime;
+            this.loggerFactory = loggerFactory;
+            this.broadcastFrequencySeconds = broadcastFrequencySeconds;
```

Comment:
Yes bring to `SignalRSettings` and inject into the constructor.

Reply:
Done, but could not see an easy way on how to do this on the constructor using this DI Framework. We had the Initialise method on the Broadcaster so used that instead